shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1765,6 +1765,13 @@ void WebContents::OnGetDefaultPrinter(
 
   base::string16 printer_name =
       device_name.empty() ? default_printer : device_name;
+
+  // If there are no valid printers available on the network, bail.
+  if (printer_name.empty() || !IsDeviceNameValid(printer_name)) {
```

Comment:
https://github.com/electron/electron/blob/master/shell/browser/api/electron_api_web_contents.cc#L1851-L1856 check is now redundant, only advantage is bailing early which doesn't save much, we can remove that.

Reply:
@deepak1556 hmm - i'm rethinking this in light of the newly failing test. I think there is in fact value in telling the user that _they_ passed in an invalid printer independent of printers being available on the network. One throws an error and the other calls a callback, and i think as a rule user-error should be surfaced early with a true Error and not just a callback.

Reply:
I'm good with either solution. But we should explicitly document that the `print` throws on invalid input, currently the API design implies that all errors are passed via callback.

Reply:
I am fine with it as well, was suggesting this because errors can be raised through callbacks.

Reply:
done!